src/auth/withAuthRedirect.tsx

```diff
@@ -1,19 +1,17 @@
-import React, {FC} from 'react';
+import React, {ComponentType, FC} from 'react';
 import {useSelector} from 'react-redux';
 import {Redirect} from 'react-router-dom';
 
 import {getAuth} from '../selectors';
 
-type PropsType = any;
-
-const withAuthRedirect = (ComponentToBeRendered: any) => {
-  const WrappedComponent: FC<PropsType> = props => {
+function withAuthRedirect<WCProps>(WrappedComponent: ComponentType<WCProps>) {
```

Comment:
Too complex name for a generator param. It looks like a real type. Something like `P` is fine